pkg/storage/tscache/skl_impl.go

```diff
@@ -126,3 +131,12 @@ func nonNil(b []byte) []byte {
 	}
 	return b
 }
+
+// checkKeyLen asserts that the key length is below the maximum key length
+// allowed by intervalSkl. This is very large (4 GB), so we just fatal if
+// we see keys of this size. We should be catching this at a higher level.
```

Comment:
Do we have a higher-level check for this? If so, where? If not, we should either add one or make this return an error rather than crashing.

Reply:
`func (r *Replica) checkBatchRequest(ba roachpb.BatchRequest, isReadOnly bool) error {`?

but actually ISTM that the right thing to do is to go into `func (r *Replica) updateTimestampCache(ba *roachpb.BatchRequest, br *roachpb.BatchResponse) {
` and check the key there. If one is too long, we raise the low water mark and log a warning.

Reply:
I've been looking into adding a limit for key lengths to address #19746 and #19751, but after talking to @bdarnell he made a good point about how this should be handled by a constraint on possible schemas.. disallowing indexes on columns whose type has an unbounded size altogether.

In light of that being a larger change than I was expecting and not coming in any time soon, I'm going to change this around a bit (similar to what @tschottdorf suggested).